modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -465,29 +461,101 @@ private void stopLeaseProlongation(ReplicationGroupId groupId, @Nullable String
         });
     }
 
+    public CompletableFuture<Replica> getReplica(ReplicationGroupId replicationGroupId) {
+        return replicas.get(replicationGroupId);
+    }
+
+    public boolean isRaftClientStarted(RaftNodeId raftNodeId) {
+        return ((Loza) raftManager).isStarted(raftNodeId);
+    }
+
+    public void resetPeers(RaftNodeId raftNodeId, PeersAndLearners peersAndLearners) {
+        ((Loza) raftManager).resetPeers(raftNodeId, peersAndLearners);
+    }
+
+    public LogSyncer getLogSyncer() {
+        return raftManager.getLogSyncer();
+    }
+
+    /**
+     * TODO.
+     *
+     * @param isVolatileStorage TODO
+     * @param volatileLogStorageFactoryCreator TODO
+     * @return TODO
+     */
+    public RaftGroupOptions createRaftGroupOptions(boolean isVolatileStorage, LogStorageFactoryCreator volatileLogStorageFactoryCreator) {
+        if (isVolatileStorage) {
+            LogStorageBudgetView view = ((Loza) raftManager).volatileRaft().logStorage().value();
+            return RaftGroupOptions.forVolatileStores()
+                    // TODO: use RaftManager interface, see https://issues.apache.org/jira/browse/IGNITE-18273
+                    .setLogStorageFactory(volatileLogStorageFactoryCreator.factory(view))
+                    .raftMetaStorageFactory((groupId, raftOptions) -> new VolatileRaftMetaStorage());
+        } else {
+            return RaftGroupOptions.forPersistentStores();
+        }
+    }
+
+    /**
+     * TODO.
+     *
+     * @param groupOptions TODO
+     * @param raftGrpLsnr TODO
+     * @param raftGrpEvtsLsnr TODO
+     * @param raftNodeId TODO
+     * @param stableConfiguration TODO
+     * @throws NodeStoppingException TODO
+     */
+    public void startPartitionRaftGroupNode(
```

Comment:
Why we need this direct method? Can we do it internally on startReplica instead? It was the main idea - incapsulate as many RAFT logic as we can to the Replica(Manager) layer

Reply:
The next 3 arguments are being created in `TableManager`:
```
RaftGroupListener raftGrpLsnr
RaftGroupEventsListener raftGrpEvtsLsnr
RaftGroupOptions groupOptions
```
Some requires storages etc.. I can pass them through another lambda -- will it better solution for the ticket purposes?